apps/token/src/app.tsx

```diff
@@ -26,9 +26,148 @@ import {
   EnvironmentProvider,
   NetworkLoader,
 } from '@vegaprotocol/environment';
-import { createClient } from './lib/apollo-client';
 import { createConnectors } from './lib/web3-connectors';
 import { ENV } from './config/env';
+import type {
+  FieldFunctionOptions,
+  InMemoryCacheConfig,
+  Reference,
+} from '@apollo/client';
+import sortBy from 'lodash/sortBy';
+import uniqBy from 'lodash/uniqBy';
+
+import { deterministicShuffle } from './lib/deterministic-shuffle';
+import { addDecimal } from '@vegaprotocol/react-helpers';
+
+const formatUintToNumber = (amount: string, decimals = 18) =>
+  addDecimal(amount, decimals).toString();
+
+const createReadField = (fieldName: string) => ({
+  [`${fieldName}Formatted`]: {
+    read(_: string, options: FieldFunctionOptions) {
+      const amount = options.readField(fieldName) as string;
+      return amount ? formatUintToNumber(amount) : '0';
+    },
+  },
+});
+
+// Create seed in memory. Validator list order will remain the same
+// until the page is refreshed.
+const VALIDATOR_RANDOMISER_SEED = (
+  Math.floor(Math.random() * 1000) + 1
+).toString();
+
+const cache: InMemoryCacheConfig = {
+  typePolicies: {
+    Query: {
+      fields: {
+        nodes: {
+          // Merge function to make the validator list random but remain consistent
+          // as the user navigates around the site. If the user refreshes the list
+          // will be randomised.
+          merge: (existing = [], incoming) => {
+            // uniqBy will take the first of any matches
+            const uniq = uniqBy([...incoming, ...existing], 'id');
+            // sort result so that the input is consistent
```

Comment:
Question (non-blocking): why is it important the the input is consistent?

Reply:
So this was a random requirement, the valdiators wanted the list to be randomised before the scoring mechanism was introduced. But we wanted to persist that across a user session so the table didn't jump around on update. This is the fix for that, when we do the validator table improvements my hope is we can kill this 🙏

Reply:
Or possibly have we already sorted the table and this can go now?